src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionServerSettings.cs

```diff
@@ -1893,42 +1867,92 @@ public bool TryReadSessionTokenIdentifier(XmlReader reader)
-            public class SecurityReplySessionServiceChannelDispatcher : ServerSecuritySimplexSessionChannel, IServiceChannelDispatcher //, IReplySessionChannel 
+            //Renamed SecurityReplySessionChannel => SecurityReplySessionServiceChannelDispatcher (implementd IServiceChannelDispatcher)
+            public class SecurityReplySessionServiceChannelDispatcher : ServerSecuritySimplexSessionChannel, IServiceChannelDispatcher, IReplySessionChannel
             {
-                private IReplyChannel incomingChannel;
+                private readonly IServiceProvider serviceProvider;
+                private IServiceChannelDispatcher channelDispatcher;
                 public SecurityReplySessionServiceChannelDispatcher(
                     SecuritySessionServerSettings settings,
                     SecurityContextSecurityToken sessionToken,
                     object listenerSecurityState, SecurityListenerSettingsLifetimeManager settingsLifetimeManager
                     , IChannel channel, EndpointAddress address)
-                    : base(settings,    
+                    : base(settings,
                           sessionToken, listenerSecurityState, settingsLifetimeManager, address)
                 {
-                    this.incomingChannel = (IReplyChannel)channel;
-            
+                    this.IncomingChannel = (IReplyChannel)channel;
+                    // this.defaultBinder = new ReplyChannelBinder();
+                    // this.idleManager = new ServiceChannel.SessionIdleManager();
+                    serviceProvider = this.IncomingChannel.GetProperty<IServiceScopeFactory>().CreateScope().ServiceProvider;
                 }
 
-                public IReplyChannel IncomingChannel
+                public IReplyChannel IncomingChannel { get; set; }
+
+                public IServiceChannelDispatcher ChannelDispatcher { get => throw new NotImplementedException(); set => throw new NotImplementedException(); }
+
+                protected override bool CanDoSecurityCorrelation => true;
+
+                public event EventHandler Closed;
+                public event EventHandler Closing;
+                public event EventHandler Faulted;
+                public event EventHandler Opened;
+                public event EventHandler Opening;
+
+                public void Abort()
                 {
-                    get { return incomingChannel; }
-                    set { incomingChannel = value; }
+                    base.AbortCore();
                 }
 
-                protected override bool CanDoSecurityCorrelation => true;
+                public Task CloseAsync()
+                {
+                    TimeoutHelper helper = new TimeoutHelper(ServiceDefaults.CloseTimeout);
+                    return CloseAsync(helper.GetCancellationToken());
+                }
+
+                public Task CloseAsync(CancellationToken token)
+                {
+                    base.CloseAsync(token);
```

Comment:
This will execute the close in the background and return that the close has completed immediately. Is this the intention?. Also I'm not seeing how this method isn't causing a compiler a warning. You aren't overriding the base class method and you don't have a new keyword to make it explicit that this is a different method.

Reply:
fixed the compiler warning.. should have paid attention...